builder/pipeline_test.cc

```diff
@@ -1017,9 +1118,21 @@ TEST(pipeline, unrelated) {
   for (int i = 0; i < N2; ++i) {
     ASSERT_EQ(out2_buf(i), 2 * i + 1);
   }
+
+  {
+    pipeline p_replica = unrelated_replica();
+
+    std::string replica_text = define_replica_pipeline(ctx, {in1, in2}, {out1, out2});
+    LOG_REPLICA_TEXT(replica_text);
+    size_t pos = replica_pipeline_test_src.find(replica_text);
+    ASSERT_NE(pos, std::string::npos) << "Matching replica text not found, expected:\n" << replica_text;
+
+    test_context eval_ctx;
+    p_replica.evaluate(inputs, outputs, eval_ctx);
```

Comment:
I was thinking that the test of these pipelines was *in* `replica_pipeline_test.cc` (was: `replica_pipeline_srcs.cc`). That would simplify this a lot. It would basically be just a one line change to each test in pipeline_test.cc, for example here it would just be:

`check_replica_pipeline(define_replica_pipeline(ctx, {in1, in2}, {out1, out2}))`

Where all `void check_replica_pipeline(const std::string&)` does is check if that replica pipeline substring exists in `replica_pipeline_test.cc`.

The actual call to `evaluate` to check the replica pipeline runs would be in `replica_pipeline_test.cc`.

I was thinking that we'd just duplicate the construction of the input/output buffers there. I was also thinking that we're probably going to want a way to automate that as part of the replica functionality, which would allow eliminating that duplication, but I was thinking that could be a separate change.

Reply:
Actually there's another catch: we also want to *execute* the pipeline in question, which means we need to know which pipeline factory to use. I'll figure something out.

Reply:
ok, PTAL